The weather integration should only use the users approximate city for querying the weather and not their exact location.

diff --git i/src/weather/gcal-weather-service.c w/src/weather/gcal-weather-service.c
index 9fbedb51..57f88160 100644
--- i/src/weather/gcal-weather-service.c
+++ w/src/weather/gcal-weather-service.c
@@ -1118,7 +1118,7 @@ gcal_weather_service_run (GcalWeatherService *self,
       g_cancellable_reset (self->location_cancellable);
 
       gclue_simple_new (APPLICATION_ID,
-                        GCLUE_ACCURACY_LEVEL_EXACT,
+                        GCLUE_ACCURACY_LEVEL_CITY,
                         self->location_cancellable,
                         (GAsyncReadyCallback) on_gclue_simple_creation_cb,
                         self);
